src/Target.h

```diff
@@ -237,6 +237,8 @@ struct Target {
     /** Given a data type, return an estimate of the "natural" vector size
      * for that data type when compiling for this Target. */
     int natural_vector_size(Halide::Type t) const {
+        user_assert(*this != Target()) << "natural_vector_size cannot be used on an uninitialized Target.\n";
```

Comment:
The wording "uninitialized" is incorrect here. Given that we use the word "unknown" in OSUnknown and ArchUnknown, I would say "unknown" target. "Default" is also acceptable.